### ISSUE TITLE
lint and tests are not working on renovate patches 

### ISSUE THREAD

**Issue by kranurag7** [SovereignCloudStack/cluster-stack-operator#108]
/kind bug

Renovate opened some patches today and we have linting and test related errors in all of them. This tracks looking into it and fixing the same. 

It's passing on main and locally as well. 

```bash
$ make lint-golang
docker run  --rm -t -i \
	-v /home/k7/go/pkg:/go/pkg \
	-v /home/k7/work/scsoss/cluster-stack-operator:/src/cluster-stack-operator \
	ghcr.io/sovereigncloudstack/cso-builder:1.1.1 lint-golang;
make: Entering directory '/src/cluster-stack-operator'
go version
go version go1.21.6 linux/amd64
golangci-lint version
golangci-lint has version 1.55.2 built with go1.21.3 from e3c2265f on 2023-11-03T12:59:25Z
GO111MODULE=on golangci-lint run -v
INFO [config_reader] Config search paths: [./ /src/cluster-stack-operator /src /]
INFO [config_reader] Used config file .golangci.yaml
INFO [lintersdb] Active 49 linters: [asasalint asciicheck bidichk bodyclose containedctx contextcheck durationcheck errchkjson errname errorlint exhaustive exportloopref forcetypeassert gci goconst gocritic godot gofmt gofumpt goimports goprintffuncname gosec gosimple govet importas ineffassign loggercheck makezero misspell nakedret nilerr noctx nolintlint nosprintfhostport prealloc predeclared reassign revive rowserrcheck staticcheck stylecheck tagliatelle thelper tparallel unconvert unused usestdlibvars wastedassign wrapcheck]
INFO [loader] Go packages loading at mode 575 (compiled_files|exports_file|files|imports|name|types_sizes|deps) took 50.978179854s
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 10.603445ms
INFO [linters_context/goanalysis] analyzers took 0s with no stages
INFO [runner] Issues before processing: 57, after processing: 0
INFO [runner] Processors filtering stat (out/in): exclude: 13/13, exclude-rules: 1/13, cgo: 57/57, path_prettifier: 57/57, skip_files: 57/57, skip_dirs: 57/57, filename_unadjuster: 57/57, autogenerated_exclude: 13/57, identifier_marker: 13/13, nolint: 0/1
INFO [runner] processing took 1.776139ms with stages: identifier_marker: 1.090082ms, autogenerated_exclude: 272.775µs, path_prettifier: 268.554µs, skip_dirs: 117.649µs, exclude-rules: 7.629µs, cgo: 7.597µs, nolint: 3.336µs, filename_unadjuster: 2.87µs, max_same_issues: 1.163µs, uniq_by_line: 631ns, severity-rules: 578ns, fixer: 509ns, skip_files: 452ns, exclude: 348ns, source_code: 330ns, max_from_linter: 323ns, max_per_file_from_linter: 308ns, sort_results: 305ns, path_prefixer: 264ns, diff: 251ns, path_shortener: 185ns
INFO [runner] linters took 918.497092ms with stages: goanalysis_metalinter: 916.571183ms
INFO File cache stats: 0 entries of total size 0B
INFO Memory: 521 samples, avg is 28.1MB, max is 138.1MB
INFO Execution took 51.914004366s
make: Leaving directory '/src/cluster-stack-operator'

$ make test-unit
CREATE_KIND_CLUSTER=true KUBEBUILDER_ASSETS="/home/k7/work/scsoss/cluster-stack-operator/hack/tools/bin/k8s/1.25.0-linux-amd64" /home/k7/work/scsoss/cluster-stack-operator/hack/tools/bin/gotestsum --junitfile=.coverage/junit.xml --format testname -- -mod=vendor \
-covermode=atomic -coverprofile=.coverage/cover.out -p=4 ./internal/controller/...
PASS internal/controller.TestMakeDiff (0.00s)
PASS internal/controller.TestGetLatestReadyClusterStackRelease/a_ready,_latest_cluster_stack_release_present (0.00s)
PASS internal/controller.TestGetLatestReadyClusterStackRelease/No_ready_cluster_stack_releases (0.00s)
PASS internal/controller.TestGetLatestReadyClusterStackRelease (0.00s)
PASS internal/controller.TestGetUsableClusterStackReleaseVersions/two_ready_cluster_stack_releases (0.00s)
PASS internal/controller.TestGetUsableClusterStackReleaseVersions/no_ready_cluster_stack_releases (0.00s)
PASS internal/controller.TestGetUsableClusterStackReleaseVersions (0.00s)
PASS internal/controller.TestGetClusterStackReleasesInSpec (0.00s)
PASS internal/controller.TestControllers (40.30s)
coverage: 50.2% of statements
PASS internal/controller (coverage: 50.2% of statements)

DONE 9 tests in 43.213s

$ make test-integration-workloadcluster
CREATE_KIND_CLUSTER=true KUBEBUILDER_ASSETS="/home/k7/work/scsoss/cluster-stack-operator/hack/tools/bin/k8s/1.25.0-linux-amd64" /home/k7/work/scsoss/cluster-stack-operator/hack/tools/bin/gotestsum --junitfile=.coverage/junit.xml --format testname -- -mod=vendor \
-covermode=atomic -coverprofile=.coverage/cover.out -p=1  ./internal/test/integration/workloadcluster/...
PASS internal/test/integration/workloadcluster.TestControllers (39.84s)
coverage: [no statements]
PASS internal/test/integration/workloadcluster

DONE 1 tests in 42.282s
```

**Comment by kranurag7** [comment]
The reason was that, we were not updating our `.bulider-image-version.txt` with renovate earlier. This should be fixed by https://github.com/SovereignCloudStack/cluster-stack-operator/pull/97

**Comment by jschoone** [comment]
> The reason was that, we were not updating our `.bulider-image-version.txt` with renovate earlier. This should be fixed by #97

Hi @kranurag7, did it fix that? Please close this issue as completed then since GitHub unfortunately doesn't do it automatically when adding the relationship later.

**Comment by kranurag7** [comment]
Yes, it was fixed by #97 I'll close the issue. I missed using Fixes while creating PR.

Thanks for remainder.
